### PR TITLE
Added content and styling to Item Details Page

### PR DESCRIPTION
**Pull request by clinturbin**
For this branch:
- I added an additional fetch request to the ComponentDidMount function to also get a list of hints for an item and store it in the local state
- I separated the ItemDetailsScreen into smaller components and saved them in the item-details-screen folder
- I applied styling for mobile and desktop layouts. Given the custom nature of the layout for this screen I didn't use the class names we have been previously using.

@running-on-sunshine: Could you please review?

### CODE REVIEW

File: frontend/src/stylesheets/item-details-screen.css [hunk 1]
```diff
@@ -1,3 +1,181 @@
...
+    }
+}
+
+@media(min-width: 1200px) {
+    .item-details-container {
+        width: 70%;
+    }
+}
+
+@media(min-height: 768px) {
```
Comment (running-on-sunshine) [inline frontend/src/stylesheets/item-details-screen.css]: On line 170, there seems to be a duplicate 768px media query.

File: frontend/src/stylesheets/item-details-screen.css [hunk 2]
```diff
@@ -1,3 +1,181 @@
...
+    .item-details-container {
+        height: 500px;
+    }
+}
+
+@media(min-height: 1200px) {
```
Comment (running-on-sunshine) [inline frontend/src/stylesheets/item-details-screen.css:176]: On line 176, it looks like there is a duplicate 1200px media query.

Comment (clinturbin) [reply frontend/src/stylesheets/item-details-screen.css:176]: Those aren't duplicates. They refer to the min-height not width.

Comment (clinturbin) [reply frontend/src/stylesheets/item-details-screen.css:176]: On iPad there was a height issue where the map was stretched too tall. By using the min-height I was able to control the height a little better without affecting the desktop display

Comment (running-on-sunshine) [reply frontend/src/stylesheets/item-details-screen.css:176]: Ah! I see the difference now. Thank you so much for providing such a thorough explanation! 😊

**Comment by running-on-sunshine** [comment]
This is absolutely wonderful work! 

Thoughts: 
- I love how you broke the ItemDetailsScreen into smaller components! Makes the file much more readable and easier to debug! 😉
- Definitely also liked how you styled the screen and gave it its own unique class names! Going forward, this is what I might like to do with the other screens as well. 🙂

You are definitely good to merge! 👍

**Review by running-on-sunshine — APPROVED** [review]

**Pull request merged**